tools/bumper-bot/bumper.go

```diff
@@ -0,0 +1,104 @@
+//usr/bin/env go run $0 "$@"; exit
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/kubevirt/cluster-network-addons-operator/tools/bumper-bot/parser"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Println("Error: must contain path to components yaml", os.Args)
+		return
+	}
+	componentsYaml := os.Args[1]
+
+	fmt.Println("parsing", componentsYaml)
+	components, err := parser.ParseComponentsYaml(componentsYaml)
+	if err != nil {
+		fmt.Println(errors.Wrap(err, "failed to parse components yaml"))
+		return
+	}
+
+	for i, component := range components {
+		printCurrentComponentParams(component)
+		currentReleaseVersion, err := getCurrentReleaseTag(component.Url)
+		if err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to get latest release version tag from %s", component.Url))
+			return
+		}
+
+		latestReleaseVersion, latestReleaseCommit, err := getLatestReleaseInfo(component.Url)
+		if err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to get latest release version tag from %s", component.Name))
+			return
+		}
+
+		//TODO make updatePolicyDummy dynamic
+		updatePolicyDummy := "static"
+		bumpNeeded, err := isBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicyDummy)
+		if err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to discover if Bump need for %s", component.Name))
+			return
+		}
+
+		if bumpNeeded {
+			fmt.Printf("bumping %s from %s to %s", component.Name, currentReleaseVersion, latestReleaseVersion)
+			//reset --hard git repo
+
+			//PR name
+			PRTitle := fmt.Sprintf("bump %s to %s", component.Name, latestReleaseVersion)
+			fmt.Println("PR title", PRTitle)
+			//Create PR
+
+			//update component's entry in config yaml
+			components[i].Commit = latestReleaseCommit
+			components[i].Release_tag = latestReleaseVersion
+			err = parser.UpdateComponentsYaml(components)
+			if err != nil {
+				fmt.Println(errors.Wrap(err, "failed to update components yaml"))
+				return
+			}
+
+			//run bump script
+			cmd := exec.Command("make", fmt.Sprintf("bump-%s", component.Name))
+			if out, err := cmd.CombinedOutput(); err != nil {
+				fmt.Println(errors.Wrapf(err, "failed to run bump script. StdOut = %s", string(out)))
+				return
+			}
+
+			//create a new branch name
+			BranchName := strings.Replace(PRTitle, " ", "_", -1)
+			fmt.Println("Opening new Branch", BranchName)
+
+			//push branch to PR
+
+		}
+	}
+}
+
+func printCurrentComponentParams(component parser.Component) {
+	fmt.Println("current component config:")
+	fmt.Println("Name:", component.Name)
+	fmt.Println("Url:", component.Url)
+	fmt.Println("Commit:", component.Commit)
+	fmt.Println("Release Tag:", component.Release_tag)
```

Comment:
You can encode it to json and print the json, that's why you don't forget any param.